src/webserver/webserver.c

```diff
@@ -687,11 +741,56 @@ void http_init(void)
 	// Add access control list if configured (last two options)
 	if(strlen(config.webserver.acl.v.s) > 0)
 	{
-		options[++next_option] = "access_control_list";
+		conf_opts[idx * 2] = strdup("access_control_list");
 		// Note: The string is duplicated by CivetWeb, so it doesn't matter if
 		//       the original string is freed (config changes) after mg_start()
 		//       returns below.
-		options[++next_option] = config.webserver.acl.v.s;
+		conf_opts[idx * 2 + 1] = strdup(config.webserver.acl.v.s);
+		idx++;
+	}
+
+	cJSON *option = NULL;
+	cJSON_ArrayForEach(option, config.webserver.advancedOpts.v.json)
+	{
+		if(!cJSON_IsString(option))
+		{
+			log_err("Invalid option in webserver.advancedOpts!");
+			continue;
+		}
+
+		// Get option value
+		const char *opt = cJSON_GetStringValue(option);
+
+		// Split option into key and value at the first '='
+		char *equal_sign = strchr(opt, '=');
+		if(equal_sign == NULL)
+		{
+			log_err("Invalid option in webserver.advancedOpts: %s (missing '=')", opt);
+			continue;
+		}
+
+		// Allocate memory for key and value
+		size_t key_len = (size_t)(equal_sign - opt);
+		char *key = calloc(key_len + 1, sizeof(char));
+		if (key == NULL) {
+			log_err("Failed to allocate memory for advanced webserver option key!");
+			continue;
+		}
+		strncpy(key, opt, key_len);
+		key[key_len] = '\0';
+
+		char *value = strdup(equal_sign + 1);
+		if (value == NULL) {
+			log_err("Failed to allocate memory for advanced webserver option value!");
+			free(key);
+			continue;
+		}
+
+		// Store key and value in options array (already allocated
+		// above)
+		conf_opts[idx * 2] = key;
+		conf_opts[idx * 2 + 1] = value;
+		idx++;
```

Comment:
I didn't meant to remove the log line completely, but the id was confusing.

```suggestion
        log_info("Using advanced webserver option: \"%s\"=\"%s\"", key, value);
		idx++;
```

Reply:
> I didn't meant to remove the log line completely, but the id was confusing.

Maybe you didn't but I actually intended to remove it entirely :wink: Reason is that I don't see any value in printing some parts of the applied config. Either print all of them or nothing IMO. And it really shouldn't matter as users can trust that FTL applies the config in exactly the way they want. We only print what FTL actually applies either in debug mode or when a fatal error occurs as it may be of relevance in this case.